effectsize_exemplar_within.Rmd

```diff
@@ -0,0 +1,137 @@
+## Exemplar: Within-subjects experiment {#effectsize_exemplar_within} 
+
+Large individual differences can be a major source of noise. An effective way of accounting for that noise is for every subject to run in every combination of conditions multiple times.
+
+In this example, we'll compare two interfaces for visualizing data.
+
+* Independent Variable **layout**: the two layouts of the interface
+* Independent Variable **size**: the size of the dataset visualized (small, medium, and large)
+* Independent Variable **color**: interface color, where we don't expect any effect
+
+We run each subject through each combination of these variables 20 times to get (2 interfaces) × (3 sizes) × (4 colors) × (20 repetitions) = `r 2*3*4*20` trials per subject. We measure some reponse (e.g., reponse time) in each trial.
+
+
+### Libraries needed for this analysis
+
+```{r setup, warning = FALSE, message = FALSE}
+library(tidyverse)
+library(broom)      # for tidy()
+library(afex)       # for aov_ez()
+```
+
+```{r boilerplate, include = FALSE}
+format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdigits, format = "fg", flag="#"))
+```
+
+
+### Subjects, conditions, and repetitions
+In this example, there are 10 subjects (`id` column). Because this is simulated data, we're using subject `id` to represent individual performance differences. Because within-subjects experiments partly account for individual differences, they often need far fewer subjects than between-subject designs. Repetitions also help reduce noise.
+
+```{r within-setup}
+set.seed(456) # make the output consistent
+data <- expand.grid(
+  id = rnorm(6, 5, 0.5), # individual differences
+  layout = 0:1, # independent variable
+  size = 0:2, # independent variable
+  color = 0:3, # independent variable
+  repetition = 1:20 # each subject runs in each condition multiple times
+)
+```
+
+### Simulate some noisy effects
+We'll simulate an experiment with a main effect of `layout` and `size` and an interaction between them. However, `color` and its interactions will not have an impact.
+
+```{r within-simulate}
+data <- 
+  data %>% 
+  mutate(
+  response_time = 
+    id + # additive individual differences
+    layout * .4 + # main effect of layout
+    size * .2 + # main effect of size
+    color * 0 + 
+    layout * size * .5 + # 2-way interaction
+    size * color * 0 + 
+    layout * color * 0 + 
+    layout * size * color * 0 + 
+    rnorm(n()) # noise
+)
+```
+
+Even though we used numbers to simulate the model, the independent variables and subject ID are all factors.
+```{r within-factor}
+data <- 
+  data %>% 
+  mutate(
+    id = factor(id), 
+    layout = factor(layout), 
+    size = factor(size), 
+    color = factor(color)
+  )
+```
+
+
```

Comment:
@steveharoz Can we show an excerpt from the data table here and then some descriptive visualizations so that people have a sense of what the data looks like?